atoma-proxy/src/server/streamer.rs

```diff
@@ -298,6 +304,27 @@ impl Stream for Streamer {
                     }
                 };
 
+                if let Some(remaining) = KEEP_ALIVE_CHUNK_STR.get(self.keep_alive_pos..) {
+                    if remaining.starts_with(chunk_str) {
+                        self.keep_alive_pos += chunk_str.len();
+
+                        if self.keep_alive_pos == KEEP_ALIVE_CHUNK_STR.len() {
+                            // Full keep-alive received
+                            self.keep_alive_pos = 0;
+                            return Poll::Pending;
+                        }
+                        return Poll::Pending;
+                    } else if self.keep_alive_pos > 0 {
+                        // Reset position if we had partial match but current chunk doesn't continue it
```

Comment:
this should never happen, the chunk (and their pieces) are ordered, so I think we should return error when this happens

Reply:
Nice observation, addressed